### DISCUSSION TITLE
Track events in a mobile app

### DISCUSSION THREAD

**Discussion by mariuswestgaard** [plausible/analytics#677] [Anything Else]
Hi!

First of all: I love your product! It's working great. The only thing "missing" for me is a way to track mobile app activity.

The website we're building has an accompanying mobile app. Is it possible to track app-events from the app using the event-api? 
If not, it would be great to have an sdk, or even just an api-endpoint to manully trigger a "pageview" event (or other predefined events) from the app.

**Update:** You can now track mobile apps using our events API: https://plausible.io/docs/events-api

**Comment by Maistho** [comment]
I think it should be possible, from reading the source it should be possible to send POST requests to /api/event

Would be nice if there was some documentation other than the source of the tracker to find out what the event API looks like :)

**Comment by mvolpato** [comment]
The linked page is returning 404. 

I would love this feature as well.

**Comment by metmarkosaric** [comment]
some people already use Plausible like this to track their mobile apps so it is technically already supported but it's just not something we've officially documented yet. writing the docs is on our to do list but cannot make any promise on when it will be live. 

just a note that if you use this method to track your mobile app, your Plausible dashboard stays the same with all the same metrics as the normal dashboard as we have no mobile-specific metrics

**Reply by appinteractive** [reply]
I Do use that already too, just tried to get a grasp from the tacking scripts and build a simple request in flutter for that. 
The post request data looks like that (in dart):


```dart    
post(
  Uri.parse('https://sub.domain.tld/api/event'), // replace that domain with your (self) hosted solution.
  headers: {
    'Accept': '*/*',
    'Content-Type': 'text/plain',
    'Host': 'sub.domain.tld', // put in here your self hosted plausible instance or the one on plausible.io
    'Origin': 'https://sub.domain.tld', // here replace that with your site attached to the dashboard (I use app.mydomain.tld)
    'User-Agent': userAgent ?? '', // this is one I struggle with currenlty
  },
  body: jsonEncode({
    'n': name, // name of the event or 'pageview' for a view event
    'u': url, // url you want to track
    'd': null, // I do not know definitly
    'r': null, // referer
    'w': screenWidth, 
    'h': 0, // screen height???
    'p': props != null ? jsonEncode(props) : null, // props you want to send with the event
  })
);

```

@mariuswestgaard might that help a bit?
All in all is thet approach so far correct? @metmarkosaric 

You will need to assign a domain to the app, I did use a subdomain which does not exist for that.

Another note is that you need to send pageview events in order to get stats, custom events are not enough as they get not countet (which makes total sense)

The only bigger question I have is, how to format the user agent in order to get proper stats in the dashboard!? @metmarkosaric mine look currently like so: `CFNetwork/1237 Darwin/20.6.0 (iPhone iOS/14.5)` but does not appear in the dashboard. Any suggestion or hints on that?

**Reply by ukutaht** [reply]
@appinteractive That looks good to me. I will document the endpoint soon but you can sort of see what the parameters mean [here](https://github.com/plausible/analytics/blob/master/lib/plausible_web/controllers/api/external_controller.ex#L60).

> The only bigger question I have is, how to format the user agent in order to get proper stats in the dashboard!? @metmarkosaric mine look currently like so: CFNetwork/1237 Darwin/20.6.0 (iPhone iOS/14.5) but does not appear in the dashboard. Any suggestion or hints on that?

This gets parsed as `Safari on iOS 14.6` when I test it locally. Are you using the Cloud version or self-hosted?

**Reply by appinteractive** [reply]
Self-Hosted. Nothing appeares in the browser/device sections except the device width which I send.

**Reply by ukutaht** [reply]
OK then it's probably caused by an outdated referrers database. An updated one will be included with the next self-hosted release

**Comment by metmarkosaric** [comment]
You can take a look at our events API now https://plausible.io/docs/events-api

**Comment by leogdion** [comment]
I went ahead and created a library to do this in Swift if you are interested:
https://github.com/brightdigit/AviaryInsights
